azurerm/resource_arm_automation_dsc_nodeconfiguration.go

```diff
@@ -0,0 +1,141 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/automation/mgmt/2015-10-31/automation"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmAutomationDscNodeConfiguration() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationDscNodeConfigurationCreateUpdate,
+		Read:   resourceArmAutomationDscNodeConfigurationRead,
+		Update: resourceArmAutomationDscNodeConfigurationCreateUpdate,
+		Delete: resourceArmAutomationDscNodeConfigurationDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"automation_account_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"content": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceArmAutomationDscNodeConfigurationCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).automationDscNodeConfigurationClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for AzureRM Automation Dsc Node Configuration creation.")
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	accName := d.Get("automation_account_name").(string)
+	content := d.Get("content").(string)
+
+	s := strings.Split(name, ".")
```

Comment:
Why are you splitting the `name`? It seems that you are not using `s[1]` in the code below.